packages/network/lib/cors.ts

```diff
@@ -71,26 +74,75 @@ export function getDomainNameFromParsedHost (parsedHost: ParsedHost) {
   return _.compact([parsedHost.domain, parsedHost.tld]).join('.')
 }
 
-export function urlMatchesOriginPolicyProps (urlStr, props) {
-  // take a shortcut here in the case
-  // where remoteHostAndPort is null
+export function urlMatchesSuperDomainOriginProps (urlStr, props) {
+  return urlMatchesSameSiteProps(urlStr, props, true)
+}
+
+export function urlMatchesOriginProps (urlStr, props) {
   if (!props) {
     return false
   }
 
-  const parsedUrl = parseUrlIntoDomainTldPort(urlStr)
+  const parsedUrl = parseUrlIntoHostProtocolDomainTldPort(urlStr)
 
   // does the parsedUrl match the parsedHost?
   return _.isEqual(parsedUrl, props)
 }
 
-export function urlOriginsMatch (url1, url2) {
+/**
+ *
+ * @param {string} urlStr - The url string
+ * @param {ParsedHostWithProtocolAndHost|undefined} props - The broken down props from parseUrlIntoHostProtocolDomainTldPort
+ * @param {boolean} [strictPortMatch] - Whether or not to make port comparison exactly match. If set to true, this function checks to see if superDomainOriginPolicies match
+ * @returns {boolean}
+ */
+export function urlMatchesSameSiteProps (urlStr: string, props?: ParsedHostWithProtocolAndHost, strictPortMatch = false) {
+  if (!props) {
+    return false
+  }
+
+  const { port: port1, subdomain: _unused1, ...parsedUrl } = parseUrlIntoHostProtocolDomainTldPort(urlStr)
+  const { port: port2, subdomain: _unused2, ...relevantProps } = props
```

Comment:
why don't we just parse the 'props' url here and combine this with the `urlOrignsMatch` function?

Reply:
We have the `urlSameSiteMatch` which is the equivalent for same site for same origin check which uses the `urlMatchesSameSiteProps` under the hood. Main reason we can't check strict equality is we have to omit the subdomain (which the same-origin check doesn't) and do all this weird logic with the port 😢